### PR TITLE
Fix for review of ssl_read_certificate_verify_parse

### PR DESCRIPTION
**Pull request by lhuang04**
Summary: Review for #184 
* Remove empty line
* Remove out-of-date comments

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Notes:
* Pull requests cannot be accepted until the PR follows the [contributing guidelines](../CONTRIBUTING.md). In particular, each commit must have at least one `Signed-off-by:` line from the committer to certify that the contribution is made under the terms of the [Developer Certificate of Origin](../dco.txt).
* This is just a template, so feel free to use/remove the unnecessary things
## Description
A few sentences describing the overall goals of the pull request's commits.

## Status
**READY/IN DEVELOPMENT/HOLD**

## Requires Backporting
When there is a bug fix, it should be backported to all maintained and supported branches.
Changes do not have to be backported if:
- This PR is a new feature\enhancement
- This PR contains changes in the API. If this is true, and there is a need for the fix to be backported, the fix should be handled differently in the legacy branch

Yes | NO  
Which branch?

## Migrations
If there is any API change, what's the incentive and logic for it.

YES | NO

## Additional comments
Any additional information that could be of interest

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog updated
- [ ] Backported

## Steps to test or reproduce
Outline the steps to test or reproduce the PR here.

### CODE REVIEW

File: library/ssl_tls13_generic.c [hunk 1]
```diff
@@ -1124,7 +1123,6 @@ static int ssl_read_certificate_verify_parse( mbedtls_ssl_context* ssl,
 
     signature_scheme = ( buf[0] << 8 ) | buf[1];
 
-    /* We currently only support ECDSA-based signatures */
     switch ( signature_scheme ) {
```
Comment (hanno-becker) [inline library/ssl_tls13_generic.c]: Style: No space after `switch`

Comment (hanno-becker) [reply library/ssl_tls13_generic.c]: Style: Some lines below are overly long.

**Review by hanno-becker — CHANGES_REQUESTED** [review]
Thank you @lhuang04 

* The code uses a case distinction for SHA-256 vs. SHA-384, which should be replaced by the use of the generic hash API `mbedtls_md_xxx()`.
* As part of this review, we should also consider `mbedtls_ssl_create_verify_structure()`, which needs some rework. For example, the labels used therein could be moved to the label list in `ssl_tls13_keys.h`, and there are some 'magic numbers' in there which should either be explained in comments or, preferably, replaced by suitable macros. Also, the function being static, it should be named `ssl_xxx()` not `mbedtls_ssl_xxx()`.
* Some style issues, e.g. spacing and overly long lines, also in `ssl_certificate_verify_coordinate()`. Could you go over all functions related to CertificateVerify and make sure they adhere to the Mbed TLS style?
* The logic around missing certificates in `ssl_certificate_verify_coordinate()` isn't clear to me. I would expect the function to either skip the write if no certificate is present, or make the check an assertion rather than a possible failure. Could you take a look?

**Comment by lhuang04** [comment]
@hanno-arm , could you elaborate your last point regarding to `ssl_certificate_verify_coordinate `? This PR handles read certificate verify for #184. [ssl_certificate_verify_coordinate](https://github.com/hannestschofenig/mbedtls/blob/tls13-prototype/library/ssl_tls13_generic.c#L586-L589) is used for write certificate verify, it can be handled in #186.
```
 I would expect the function to either skip the write if no certificate is present, or make the check an assertion rather than a possible failure.
 ```

**Comment by lhuang04** [comment]
@hanno-arm , could you check again? I should have addressed most of the review comments except those for #186.

File: library/ssl_tls13_generic.c [hunk 2]
```diff
@@ -494,56 +494,53 @@ void mbedtls_ssl_set_outbound_transform( mbedtls_ssl_context *ssl,
...
                                         size_t *verify_buffer_len,
                                         int from )
 {
-    /* The length of context_string_[client|server] is
-     * sizeof( "TLS 1.3, xxxxxx CertificateVerify" ) - 1, i.e. 33 bytes.
-     */
-    const unsigned int content_string_len = sizeof( "TLS 1.3, xxxxxx CertificateVerify" ) - 1;
-    const unsigned char context_string_client[] = "TLS 1.3, client CertificateVerify";
-    const unsigned char context_string_server[] = "TLS 1.3, server CertificateVerify";
+    size_t i = 64;
```
Comment (hanno-becker) [inline library/ssl_tls13_generic.c]: Give this a descriptive name. E.g.:
```c
uint8_t const verify_padding_val = 64;
size_t const verify_padding_len = 32;
size_t i = 0;
memset( verify_buffer, verify_padding_val, verify_padding_len );
i += verify_padding_len;
```

File: library/ssl_tls13_generic.c [hunk 3]
```diff
@@ -494,56 +494,53 @@ void mbedtls_ssl_set_outbound_transform( mbedtls_ssl_context *ssl,
...
+ * The structure is computed per TLS 1.3 specification as:
+ *   - 64 bytes of octet 32,
+ *   - 33 bytes for the context string
+ *        (which is either "TLS 1.3, client CertificateVerify"
+ *         or "TLS 1.3, server CertificateVerify"),
+ *   - 1 byte for the octet 0x0, which servers as a separator,
+ *   - 32 or 48 bytes for the Transcript-Hash(Handshake Context, Certificate)
+ *     (depending on the size of the transcript_hash)
+ *
+ * This results in a total size of
```
Comment (hanno-becker) [inline library/ssl_tls13_generic.c]: This calculation should be moved to the definition of `MBEDTLS_SSL_VERIFY_STRUCT_MAX_SIZE`, which should be referred here.

File: library/ssl_tls13_generic.c [hunk 4]
```diff
@@ -1124,8 +1103,7 @@ static int ssl_read_certificate_verify_parse( mbedtls_ssl_context* ssl,
 
     signature_scheme = ( buf[0] << 8 ) | buf[1];
 
-    /* We currently only support ECDSA-based signatures */
-    switch ( signature_scheme ) {
+    switch( signature_scheme ) {
```
Comment (hanno-becker) [inline library/ssl_tls13_generic.c]:
```suggestion
    switch( signature_scheme ) 
    {
```

File: library/ssl_tls13_keys.h [hunk 5]
```diff
@@ -75,6 +77,9 @@ extern const struct mbedtls_ssl_tls1_3_labels_struct mbedtls_ssl_tls1_3_labels;
     mbedtls_ssl_tls1_3_labels.LABEL,              \
     sizeof(mbedtls_ssl_tls1_3_labels.LABEL)
```
Comment (hanno-becker) [inline library/ssl_tls13_keys.h]: Minor: Move this past the new def'n of `MBEDTLS_SSL_TLS1_3_LBL_LEN` and reuse the latter here.

File: library/ssl_tls13_generic.c [hunk 6]
```diff
@@ -660,11 +657,10 @@ int mbedtls_ssl_certificate_verify_process( mbedtls_ssl_context* ssl )
 static int ssl_certificate_verify_coordinate( mbedtls_ssl_context* ssl )
 {
     int have_own_cert = 1;
-#if defined(MBEDTLS_SHA256_C) || defined(MBEDTLS_SHA512_C)
     int ret;
-#endif /* MBEDTLS_SHA256_C || MBEDTLS_SHA512_C */
 
-    if( ssl->session_negotiate->key_exchange != MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA )
+    if( ssl->session_negotiate->key_exchange !=
```
Comment (hanno-becker) [inline library/ssl_tls13_generic.c]: Here we can use the helper `mbedtls_ssl_tls13_key_exchange_with_psk()` instead.

**Review by hanno-becker — CHANGES_REQUESTED** [review]
Thank you @lhuang04 for the changes!

A few smaller points remain as indicated in the comments.

**Comment by lhuang04** [comment]
@hanno-arm , addressed review comments, could you check again?

File: library/ssl_tls13_generic.c (hunk 2 above)
Comment (hanno-becker) [reply library/ssl_tls13_generic.c]: @lhuang04 Sorry, I might not have been clear: I meant to give a name to the `64`, not necessarily change the name of `i`.

**Review by hanno-becker — APPROVED** [review]

**Pull request merged**